### PR TITLE
JBPM-5333 Redundant modal shown when exception is thrown during start…

### PR DESCRIPTION
**Pull request by bxf12315**
…ing of process

@cristianonicolai Please review it.

### CODE REVIEW

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

File: jbpm-wb-forms/jbpm-wb-forms-client/src/main/java/org/jbpm/workbench/forms/client/display/process/AbstractStartProcessFormDisplayer.java [hunk 1]
```diff
@@ -203,7 +203,7 @@ public boolean error(Message message,
                                  Throwable throwable) {
                 String notification = Constants.INSTANCE.UnexpectedError(throwable.getMessage());
                 errorPopup.showMessage(notification);
-                return true;
+                return false;
```
Comment (cristianonicolai) [inline jbpm-wb-forms/jbpm-wb-forms-client/src/main/java/org/jbpm/workbench/forms/client/display/process/AbstractStartProcessFormDisplayer.java]: @bxf12315 have you considered removing _getUnexpectedErrorCallback_ ? I think that could be a good approach to fallback to the default error handler instead ( _DefaultWorkbenchErrorCallback_ ), wdyt?

File: jbpm-wb-forms/jbpm-wb-forms-client/src/main/java/org/jbpm/workbench/forms/client/display/KieWorkbenchFormsStartProcessDisplayer.java [hunk 2]
```diff
@@ -58,8 +58,7 @@ public IsWidget getFormWidget() {
     @Override
     public void startProcessFromDisplayer() {
         if (formRenderer.isValid()) {
-            service.call(getStartProcessRemoteCallback(),
-                         getUnexpectedErrorCallback()).startProcessFromRenderContext(
```
Comment (cristianonicolai) [inline jbpm-wb-forms/jbpm-wb-forms-client/src/main/java/org/jbpm/workbench/forms/client/display/KieWorkbenchFormsStartProcessDisplayer.java]: @bxf12315 can _getUnexpectedErrorCallback_ method be completely removed?

Comment (bxf12315) [reply jbpm-wb-forms/jbpm-wb-forms-client/src/main/java/org/jbpm/workbench/forms/client/display/KieWorkbenchFormsStartProcessDisplayer.java]: @cristianonicolai  I think, it is ok, In here , just have two class call getUnexpectedErrorCallback of AbstractStartProcessFormDisplayer.

**Comment by bxf12315** [comment]
@jsoltes For the test, I will take a try.

**Comment by bxf12315** [comment]
@cristianonicolai I added a test for it. But I am no sure , is it reasonable, please give me some suggestion.

File: jbpm-wb-forms/jbpm-wb-forms-client/src/main/java/org/jbpm/workbench/forms/client/display/KieWorkbenchFormsStartProcessDisplayer.java (hunk 2 above)
Comment (cristianonicolai) [reply jbpm-wb-forms/jbpm-wb-forms-client/src/main/java/org/jbpm/workbench/forms/client/display/KieWorkbenchFormsStartProcessDisplayer.java]: The compilation is failing now, please take another look.

File: jbpm-wb-forms/jbpm-wb-forms-client/src/test/java/org/jbpm/workbench/forms/client/display/KieWorkbenchFormsStartProcessDisplayerTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        processDisplayerConfig.setRenderingSettings(formRenderingSettings);
+
+        kieWorkbenchFormsStartProcessDisplayer = new KieWorkbenchFormsStartProcessDisplayer(formRenderer, service);;
+        when(formRenderer.isValid()).thenReturn(true);
+        when(formRenderingSettings.getTimestamp()).thenReturn(100000l);
+
+        MapModelRenderingContext mapModelRenderingContext = mock(MapModelRenderingContext.class);
+        when(formRenderingSettings.getRenderingContext()).thenReturn(mapModelRenderingContext);
+        when(mapModelRenderingContext.getModel()).thenReturn(null);
+        when(service.call(any())).thenReturn(kieWorkbenchFormsEntryPoint);
```
Comment (cristianonicolai) [inline jbpm-wb-forms/jbpm-wb-forms-client/src/test/java/org/jbpm/workbench/forms/client/display/KieWorkbenchFormsStartProcessDisplayerTest.java]: Please use _CallerMock_ instead, see example here https://github.com/kiegroup/jbpm-wb/blob/master/jbpm-wb-forms/jbpm-wb-forms-client/src/test/java/org/jbpm/workbench/forms/client/display/displayers/pr/AbstractStartProcessFormDisplayerTest.java#L55

File: jbpm-wb-forms/jbpm-wb-forms-client/src/test/java/org/jbpm/workbench/forms/client/display/KieWorkbenchFormsStartProcessDisplayerTest.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    @Test
+    public void testStartProcessFromDisplayer(){
+
+        kieWorkbenchFormsStartProcessDisplayer.initConfigs(processDisplayerConfig, onClose, onRefresh);
+        kieWorkbenchFormsStartProcessDisplayer.startProcessFromDisplayer();
+
+        verify(service).call(any());
+        verify(kieWorkbenchFormsEntryPoint).startProcessFromRenderContext(any(),
```
Comment (cristianonicolai) [inline jbpm-wb-forms/jbpm-wb-forms-client/src/test/java/org/jbpm/workbench/forms/client/display/KieWorkbenchFormsStartProcessDisplayerTest.java]: would be good if you match the expected values in here. I would recommend bringing some of the values you defined in init to here. Those are execution/parameter specific, init should generally be a setup only.

**Comment by bxf12315** [comment]
@cristianonicolai I update it . please review it.

File: jbpm-wb-forms/jbpm-wb-forms-client/src/test/java/org/jbpm/workbench/forms/client/display/displayers/pr/AbstractStartProcessFormDisplayerTest.java [hunk 5]
```diff
@@ -53,13 +52,16 @@
     @Before
     public void setupMocks() {
         processServiceCaller = new CallerMock<ProcessService>(processService);
-        getStartProcessFormDisplayer().setProcessService(processServiceCaller);
     }
 
     @Test
     public void testNotificationOnStartProcess() {
+        getStartProcessFormDisplayer().setNewProcessInstanceEvent(newProcessInstanceEvent);
```
Comment (cristianonicolai) [inline jbpm-wb-forms/jbpm-wb-forms-client/src/test/java/org/jbpm/workbench/forms/client/display/displayers/pr/AbstractStartProcessFormDisplayerTest.java]: these set service methods should be in the _setupMocks_ method.

File: jbpm-wb-forms/jbpm-wb-forms-client/src/test/java/org/jbpm/workbench/forms/client/display/KieWorkbenchFormsStartProcessDisplayerTest.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    @Test
+    public void testStartProcessFromDisplayer() {
+        when(formRenderer.isValid()).thenReturn(true);
+        when(formRenderingSettings.getTimestamp()).thenReturn(100000l);
+
+        MapModelRenderingContext mapModelRenderingContext = mock(MapModelRenderingContext.class);
+        when(formRenderingSettings.getRenderingContext()).thenReturn(mapModelRenderingContext);
+        when(mapModelRenderingContext.getModel()).thenReturn(null);
```
[the remark this answers is not in the document]
Comment (bxf12315) [reply jbpm-wb-forms/jbpm-wb-forms-client/src/test/java/org/jbpm/workbench/forms/client/display/KieWorkbenchFormsStartProcessDisplayerTest.java]: yes

**Comment by jsoltes** [comment]
jenkins please execute full downstream build

File: jbpm-wb-forms/jbpm-wb-forms-client/src/test/java/org/jbpm/workbench/forms/client/display/displayers/pr/AbstractStartProcessFormDisplayerTest.java (hunk 5 above)
Comment (cristianonicolai) [reply jbpm-wb-forms/jbpm-wb-forms-client/src/test/java/org/jbpm/workbench/forms/client/display/displayers/pr/AbstractStartProcessFormDisplayerTest.java]: if you override _setupMocks_ in the child class you can handle that. So execute what is _init_ today and then the parent method.

**Pull request merged**